frontend/packages/console-app/src/components/detect-cluster/cluster.ts

```diff
@@ -2,9 +2,17 @@ import * as React from 'react';
 // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
 // @ts-ignore: FIXME missing exports due to out-of-sync @types/react-redux version
 import { useDispatch } from 'react-redux';
-import { setActiveCluster } from '@console/internal/actions/ui';
+import { useLocation } from 'react-router-dom';
+import isMultiClusterEnabled from '@console/app/src/utils/isMultiClusterEnabled';
+import { setActiveCluster, formatNamespaceRoute } from '@console/internal/actions/ui';
+import { getCluster } from '@console/internal/components/utils/link';
+import { history } from '@console/internal/components/utils/router';
+// import { useActiveNamespace } from '@console/shared';
 import { LAST_CLUSTER_USER_SETTINGS_KEY } from '@console/shared/src/constants';
 import { useUserSettings } from '@console/shared/src/hooks/useUserSettings';
+import store from '../../../../../public/redux';
```

Comment:
Add the store isn't the best option, but there were weird bugs when using `useActiveNamespace`.  The bug is most notable when going to the pods list page for all projects and then clicking on a pod.  Instead of going to the details page, the user was routed back to the list page.

Reply:
Good question.  I thought about that last night.  I'll check it out.

Reply:
First changed the path to the store to `import store from '@console/internal/redux';` so at least it isn't relative.

So neither the `useSelector` nor the `activeNamespace` from `useActiveNamespace` worked.  Here is what I think is going on.  We need to know the current Namespace inside an useEffect hook, but since `useSelector` and `useActiveNamespace` are hooks and React gets grumppy when trying to put a hook in hook, I need to put them outside of the useEffect block.  I'm thinking that when I do that the namespace from the hook is a step behind when calling the useEffect.  

So not sure what to do.